<a href="https://www.kaggle.com/code/khadijaismail06/airbnb-paris-listings-reviews-eda?scriptVersionId=338078782" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
# importing library
import pandas as pd
pd.set_option("display.max_columns", None)

In [2]:
#reading listings data
listings = pd.read_csv(r"/kaggle/input/datasets/mysarahmadbhat/airbnb-listings-reviews/Airbnb Data/Listings.csv",encoding="ISO-8859-1",low_memory=False)
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 279712 entries, 0 to 279711
Data columns (total 33 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   listing_id                   279712 non-null  int64  
 1   name                         279537 non-null  object 
 2   host_id                      279712 non-null  int64  
 3   host_since                   279547 non-null  object 
 4   host_location                278872 non-null  object 
 5   host_response_time           150930 non-null  object 
 6   host_response_rate           150930 non-null  float64
 7   host_acceptance_rate         166625 non-null  float64
 8   host_is_superhost            279547 non-null  object 
 9   host_total_listings_count    279547 non-null  float64
 10  host_has_profile_pic         279547 non-null  object 
 11  host_identity_verified       279547 non-null  object 
 12  neighbourhood                279712 non-null  object 
 13 

## Cleaning the Data

In [3]:
# changing data type of "host_since" column to datetime
listings["host_since"] = pd.to_datetime(listings["host_since"])
# viewing the summary statistics of the numerical columns 
listings.describe()

,listing_id,host_id,host_since,host_response_rate,host_acceptance_rate,host_total_listings_count,latitude,longitude,accommodates,bedrooms,price,minimum_nights,maximum_nights,review_scores_rating,review_scores_accuracy,review_scores_cleanliness,review_scores_checkin,review_scores_communication,review_scores_location,review_scores_value
count,2.797120e+05,2.797120e+05,279547,150930.000000,166625.000000,279547.000000,279712.000000,279712.000000,279712.000000,250277.000000,279712.000000,279712.000000,2.797120e+05,188307.000000,187999.000000,188047.000000,187941.000000,188025.000000,187937.000000,187927.000000
mean,2.638196e+07,1.081658e+08,2016-04-11 13:01:44.994151680,0.865939,0.827168,24.581612,18.761862,12.595075,3.288736,1.515509,608.792737,8.050967,2.755860e+04,93.405195,9.565476,9.312869,9.701534,9.698593,9.633994,9.335364
min,2.577000e+03,1.822000e+03,2008-08-12 00:00:00,0.000000,0.000000,0.000000,-34.264400,-99.339630,0.000000,1.000000,0.000000,1.000000,1.000000e+00,20.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000
25%,1.384462e+07,1.720656e+07,2014-07-01 00:00:00,0.900000,0.780000,1.000000,-22.964390,-43.198040,2.000000,1.000000,75.000000,1.000000,4.500000e+01,91.000000,9.000000,9.000000,10.000000,10.000000,9.000000,9.000000
50%,2.767098e+07,5.826911e+07,2016-02-14 00:00:00,1.000000,0.980000,1.000000,40.710785,2.382780,2.000000,1.000000,150.000000,2.000000,1.125000e+03,96.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000
75%,3.978485e+07,1.832853e+08,2018-04-09 00:00:00,1.000000,1.000000,4.000000,41.908610,28.986730,4.000000,2.000000,474.000000,5.000000,1.125000e+03,100.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000
max,4.834353e+07,3.901874e+08,2021-02-26 00:00:00,1.000000,1.000000,7235.000000,48.904910,151.339810,16.000000,50.000000,625216.000000,9999.000000,2.147484e+09,100.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000
std,1.442576e+07,1.108570e+08,NaN,0.283744,0.289202,284.041143,32.560343,73.081309,2.133379,1.153080,3441.826611,31.518946,7.282875e+06,10.070437,0.990878,1.146072,0.867434,0.886884,0.833234,1.042625


In [4]:
# dropping rows with price and accommodates columns having zero values
listings = listings[(listings["price"] != 0) & (listings["accommodates"] != 0)]
listings.shape

(279599, 33)

There were 113 entries with zero values.

In [5]:
# checking for duplicates entries
listings.duplicated().sum()

np.int64(0)

So,there are not any duplicated rows of listings

In [6]:
# city analysis
listings["city"].value_counts()

city
Paris             64628
New York          36984
Sydney            33630
Rome              27641
Rio de Janeiro    26608
Istanbul          24517
Mexico City       20058
Bangkok           19360
Cape Town         19086
Hong Kong          7087
Name: count, dtype: int64

Paris have the most listing entries we will be analyzing paris listings data

## Paris Listings Analysis

In [7]:
# Paris listings
paris_listings = listings[listings["city"] == "Paris"][["listing_id", "host_since","neighbourhood","city","room_type","price", "review_scores_rating","review_scores_cleanliness","review_scores_location"]]
# checking for missing values in Paris listings
paris_listings.isna().sum()

listing_id                       0
host_since                      33
neighbourhood                    0
city                             0
room_type                        0
price                            0
review_scores_rating         16592
review_scores_cleanliness    16630
review_scores_location       16657
dtype: int64

In [8]:
# lets drop only host_since null values
paris_listings.dropna(subset=["host_since"], inplace=True)
paris_listings.shape

(64595, 9)

In [9]:
paris_listings.groupby("neighbourhood")["price"].mean().sort_values(ascending=False)

neighbourhood
Elysee                 211.373651
Louvre                 175.754448
Passy                  161.290875
Palais-Bourbon         157.514464
Luxembourg             155.794589
Bourse                 149.633577
Hotel-de-Ville         144.515228
Temple                 138.429300
Pantheon               122.983599
Opera                  119.201841
Vaugirard              107.086207
Enclos-St-Laurent      103.122590
Batignolles-Monceau    102.686778
Observatoire           101.996775
Gobelins                98.110184
Popincourt              90.548168
Reuilly                 89.125566
Buttes-Montmartre       87.246196
Buttes-Chaumont         82.690182
Menilmontant            74.931521
Name: price, dtype: float64

We have Elysee most expensive area in Paris with average price of €211 while Menilmontant has the cheapest average rate of €75.

In [10]:
# Yearly insights on listings
paris_listings["year"] = paris_listings["host_since"].dt.year
paris_listings.groupby("year").agg({"listing_id": "count", "price": "mean"}).sort_values("price", ascending=False)

,listing_id,price
year,,
2009,106,159.641509
2020,3363,143.517098
2018,4294,138.209362
2019,5685,129.962533
2010,416,125.031250
2011,1339,124.828230
2016,8867,114.211345
2012,4592,111.578615
2017,4585,108.658888


**Airbnb Market throughout the years** 
- Paris has lowest listings only 4 which indicates the start of bnb in paris in 2008 which grdaually increases upto 2015 maximum listings of 12147 which shows new hosts joining the bnb rise of bnb market.
- From 2015 paris listings starts decreasing showing a very huge drop due to safety concerns terrorist attacks on Paris.
- Then from 2018 bnb market starts rising again until 2020 and then it hits a massive drop again in 2021 cause of cornona virus and start of pandemic.

__Insights on average Airbnb prices over the years__
- __2008-2009:__ Prices at the start were very low avg of €78 per night but in 2009 prices got almost double €160 per night the highest increase in prices maximum price per night over the years peak of Airbnb market.
- __2010-2014:__ From 2010 prices starts decreasing with avg price falling from €125 to €100 approx.It shows increasing competition as more hosts starts entered the market during those years.
- __2015-2020:__ During those years prices gradually fall and rise until start of corona virus.
- __2021 onwards:__ Start of pandemic ban on tourism with avg of €93 per night biggest drop in prices.

In [11]:
paris_listings.groupby("room_type").agg({"listing_id": "count", "price": "mean", "review_scores_cleanliness": "mean"}).sort_values("price", ascending=False)

,listing_id,price,review_scores_cleanliness
room_type,,,
Hotel room,1334,239.633433,9.470660
Entire place,55346,114.110324,9.194027
Private room,7491,87.807903,9.273408
Shared room,424,47.205189,9.038462


__Room_type analysis__ 
- We see that almost 80% of the listings are of __Entire place__. 
- As expected hotel rooms are most expensive approx double the price of entire place cause of more services cleanliness rate higher than the others.
- Shared room have least cleanliness rate also the least avg price per night.

In [12]:
paris_listings.groupby("neighbourhood")[["price","review_scores_cleanliness","review_scores_location","review_scores_rating"]].mean().sort_values("review_scores_rating", ascending=False)

,price,review_scores_cleanliness,review_scores_location,review_scores_rating
neighbourhood,,,,
Popincourt,90.548168,9.196920,9.747535,93.615053
Enclos-St-Laurent,103.122590,9.222590,9.584301,93.471985
Menilmontant,74.931521,9.200357,9.446148,93.428622
Hotel-de-Ville,144.515228,9.306720,9.954727,93.365935
Buttes-Montmartre,87.246196,9.205104,9.401999,93.315400
Reuilly,89.125566,9.220492,9.687437,93.247614
Palais-Bourbon,157.514464,9.269753,9.903355,93.238818
Opera,119.201841,9.241556,9.816082,93.197863
Buttes-Chaumont,82.690182,9.156431,9.450995,93.081979


__Neighbourhood price relation with cleanliness and location__
- __Best neighbourhood__ Hotel-de-Ville has the highest cleanliness and location ratings despite being mid priced.
- Cheap doesn't mean badly Menilmontant has the mid level cleanliness rate while Elysee being the most expensive has low level cleanliness rate and ratings.
- Prices are most likely driven by the location not by the cleanliness.
- Popincourt is the best neighbourhood to live as it has highest ratings.

## Paris Reviews

In [13]:
#reading reviews data 
reviews = pd.read_csv(r"/kaggle/input/datasets/mysarahmadbhat/airbnb-listings-reviews/Airbnb Data/Reviews.csv", encoding="latin-1", low_memory=False)
paris_reviews = pd.merge(paris_listings,reviews, on="listing_id")
#converting date column to datetime format and extracting month from it
paris_reviews["date"] = pd.to_datetime(paris_reviews["date"])
paris_reviews["month"] = paris_reviews["date"].dt.month
paris_reviews.groupby("month")["review_id"].count().sort_values(ascending=False)


month
10    126377
7     123753
9     120990
6     118223
1     101266
8     100901
5      93975
11     91265
12     89145
4      85273
3      83827
2      77936
Name: review_id, dtype: int64

__Insights on monthly reviews__
- __Jun-Oct:__ Greatest number of reviews during this period showing peak tourism season in Paris leads to more bookings.
- __Feb-April:__ Winter having least number of reviews less tourists interaction during this season.
- Monthly stays could reflect a lag cause reviews are left after their stay so october reviews may be reflecting september stays.